beard_server/modules/clustering/views.py

```diff
@@ -0,0 +1,100 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Inspire.
+# Copyright (C) 2016 CERN.
+#
+# Inspire is free software; you can redistribute it
+# and/or modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# Inspire is distributed in the hope that it will be
+# useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Inspire; if not, write to the
+# Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+# MA 02111-1307, USA.
+#
+# In applying this license, CERN does not
+# waive the privileges and immunities granted to it by virtue of its status
+# as an Intergovernmental Organization or submit itself to any jurisdiction.
+
+"""Clustering endpoint."""
+
+from __future__ import (
+    absolute_import,
+    division,
+    print_function,
+    unicode_literals)
+
+from flask import (
+    abort,
+    Blueprint,
+    jsonify,
+    request)
+
+from .beard import (
+    predict as clustering,
+    train as learn_model)
+
+blueprint = Blueprint(
+    'beard_server_clustering',
+    __name__,
+    url_prefix='/api/clustering',
+)
+
+
+@blueprint.route('/', methods=['GET'])
+def available_methods():
+    """List of available methods to call as the part of API."""
+    available = [
+        "[POST] /api/clustering/clusters",
+        "[POST] /api/clustering/train",
+    ]
+
+    return jsonify(available)
```

Comment:
I'm not entirely convinced by this HATEOAS-like endpoint, because it doesn't communicate what the JSON sent with the request must look like: one would have to read the documentation anyway...